src/App.tsx

```diff
@@ -16,80 +19,171 @@ export const goodsFromServer = [
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  resetButton = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+
+    const preparedGoods = (
+      getReorderedGoods(goodsFromServer, { sortType, isReversed })
+    );
+
+    return (
+      <>
+        <div className="container">
+          <Box
+            sx={{
+              display: 'flex',
+              '& > :not(style)': {
+                m: 1,
+                mt: 30,
+                mx: 'auto',
+                width: 665,
+                height: 500,
+              },
+            }}
+          >
+            <Paper
+              variant="outlined"
+              sx={{
+                backgroundColor: 'text.disabled',
+                display: 'flex',
+                alignItems: 'center',
+                justifyContent: 'start',
+                fontSize: 24,
+                fontWeight: 'bold',
+              }}
+            >
+              <div className="section content">
+                <div className="buttons">
+                  <Button
+                    variant="contained"
+                    type="button"
+                    className={sortType === SortType.ALPHABET
+                      ? 'button is-info'
+                      : 'button is-info is-light'}
+                    onClick={this.sortByAlphabet}
+                  >
+                    Sort alphabetically
+                  </Button>
+
+                  <Button
+                    variant="contained"
+                    type="button"
+                    className={sortType === SortType.LENGTH
+                      ? 'button is-success'
+                      : 'button is-success is-light'}
+                    onClick={this.sortByLength}
+                  >
+                    Sort by length
+                  </Button>
+
+                  <Button
+                    variant="contained"
+                    type="button"
+                    className={isReversed
+                      ? 'button is-warning'
+                      : 'button is-warning is-light'}
+                    onClick={this.goodsReverse}
+                  >
+                    Reverse
+                  </Button>
+
+                  {sortType !== SortType.NONE || isReversed
+                    ? (
+                      <Button
+                        variant="outlined"
+                        color="error"
+                        type="button"
+                        className="button is-danger"
+                        onClick={this.resetButton}
+                      >
+                        Reset
+                      </Button>
+                    )
+                    : ''}
```

Comment:
you shouldn't render this part if it is not on a page) Try to use 

{sortType !== SortType.NONE || isReversed && (...

Reply:
Thanks for great advice, i didn't think about this!